### DISCUSSION TITLE
Disabling the /web demo page

### DISCUSSION THREAD

**Discussion by DarthVi** [umputun/remark42#1768] [Q&A] [answered]
Is there any way to disable the demo page running on a subdomain like "remark42.domain.com/web" without breaking the retrieval of the embed.js used to show the comments? I use Caddy as reverse proxy system.

**Comment by paskal** [comment, marked as answer]
I think you could just redirect root of `/web/` to another location but keep separate rules for files which are required: `remark.mjs`, `remark.css`, `iframe.html` and `embed.mjs`.

**Reply by DarthVi** [reply]
I've tried that right now, I am able to access the files but then on the website I use remark42 on I get this

![image](https://github.com/umputun/remark42/assets/8702259/d3e924da-196e-4763-846e-66ba2cb3496d)

**Reply by paskal** [reply]
What do you see on the network tab? Are any resources with `mjs` return HTML instead?

**Reply by DarthVi** [reply]
On the network tab there are no errors, everything is retrieved properly

**Reply by paskal** [reply]
Can you give me the URL to see? Lack of errors doesn't mean that there is no problem: redirect might be to HTML page and then instead of JS or CSS file you would receive an HTML page and nothing will work, which seems to be the case.

**Reply by DarthVi** [reply]
You can try here: https://darthvi.com/post/fantasy-name-generator/

**Reply by DarthVi** [reply]
Ok, I was able to fix it. Basically I was redirecting only the traffic to the aforementioned files, but not the ones about the api calls (/api/v1, find etc.). Now I set up Caddy properly and it works! I'll flag your comment as the answer because it originally was corret and the mistake was on my side.

Thank you very much!

**Reply by paskal** [reply]
Could you please post the correct config you use? So that others would be able to benefit from it.

**Reply by DarthVi** [reply]
Here it is, of course anyone that wants to benefit from it, should replace the domain and the REMARK_PORT with the appropriate custom configurations

```
remark42.domain.com {
  handle /web* {
        error 403
  }

  handle /web/remark.mjs {
        reverse_proxy localhost:REMARK_PORT
  }
  handle /web/embed.mjs {
        reverse_proxy localhost:REMARK_PORT
  }
  handle /web/iframe.html {
        reverse_proxy localhost:REMARK_PORT
  }
  handle /web/remark.css {
        reverse_proxy localhost:REMARK_PORT
  }

  reverse_proxy localhost:REMARK_PORT

  log {
    output file /var/log/caddy/remark-access.log {
        roll_size 20mb
        roll_keep 10
        roll_keep_for 720h
    }
  }
}
```

**Reply by paskal** [reply]
Unfortunately GitHub login on the URL you provided won't work, [here](https://remark42.com/docs/manuals/separate-domain/)  are the details of the limitations of a setup on a separate domain.

**Reply by DarthVi** [reply]
But at least for my account it is working (I haven't tested with other accounts, I may ask a friend of mine to try to login). I've set ALLOWED_HOSTS to the domains where the blog is deployed and AUTH_SAME_SITE to 'none'.

I've also added the Github .svg paths to the Caddyfile handles, so now even the icon works.

**Reply by DarthVi** [reply]
I can confirm that a friend of mine was able to login via Github and write a comment. He used Firefox.
I noticed that login is not possible if the user uses Incognito Mode on Chrome, since in that case Chrome by default does not allow websites to store third-party cookies (a thing that anyway can be enabled via Chrome's Setting preferences).

**Reply by paskal** [reply]
It doesn't work on Safari either it seems.

**Reply by DarthVi** [reply]
It probably is due to a third-party cookies setting even for Safari. I can see [here](https://macpaw.com/how-to/allow-third-party-cookies-mac) that by default Safari blocks cross site (third party) cookies.

The aforementioned friend of mine who also owns a Mac made a test, and I can confirm what I've written in the previous lines: with the appropriate settings, even Safari works.
This is a problem that even systems like disqus should have, since their cookies come form disqus.com and not from the blog/site using them.
